editoast/src/views/path/projection.rs

```diff
@@ -437,20 +437,68 @@ mod tests {
         #[case] path: &[&str],
         #[case] track_ranges: &[&str],
         #[case] expected_intersections: &[(u64, u64)],
+        #[values(false, true)] toggle_path: bool,
+        #[values(false, true)] toggle_track_ranges: bool,
     ) {
-        let path: Vec<TrackRange> = path.into_iter().map(|s| s.parse().unwrap()).collect();
+        let path: Vec<TrackRange> = path
+            .into_iter()
+            .map(|s| {
+                let mut track_range: TrackRange = s.parse().unwrap();
+                if toggle_path {
+                    track_range.direction = track_range.direction.toggle();
+                }
+                track_range
+            })
+            .collect();
+        let path = if toggle_path {
+            path.into_iter().rev().collect()
+        } else {
+            path
+        };
```

Comment:
Just a reminder from our call: some of the code here could be moved into a separate function for reuse.

Reply:
I've done [5969536](https://github.com/OpenRailAssociation/osrd/pull/10137/commits/59695360472393d452051d0d5a01c09faf8f0483) and [ba541fa](https://github.com/OpenRailAssociation/osrd/pull/10137/commits/ba541faed7d0ee2d3968e4c27878fc6621a09d6a) to extract some functions and make the test function itself, shorter and more explicit about what it does. What do you think?

Reply:
Looks good, thanks!